core/update.php

```diff
@@ -545,6 +549,7 @@ function update_check_requirements($skip_warnings = FALSE) {
       }
 
     case t('Apply pending updates'):
+      update_upgrade_enable_dependencies();
```

Comment:
This perhaps should be moved _before_ the "Apply updates" button? The `update_fix_compatibility()` function does the exact opposite of this function: it _disables_ old modules. This function enables new ones. Perhaps the two of them could be either named similar or be merged together some how?

By moving up to above, before the updates are applied, any installation requirements that are not met by the newly installed modules will be displayed by `update_check_requirements()`. For example if a newly installed module required a PHP extension that we not available, that should be installed before attempting to upgrade the site, as update hooks may depend on the extension.

Reply:
@quicksketch ,  I see the logic of your comment. One of the reason I decided to place this call later in the upgrade process was that the user would have a chance to see what modules would be automatically updated before proceeding with the updates, giving them a chance to cancel. This was suggested in a comment by @docwilmot. If I move `update_upgrade_enable_dependency()` before `update_fix_compatibility()` then this option is gone.

If you are OK with this, I'll move it as requested. Let me know.